extensions/wikia/Chat2/ChatBanListSpecial_helper.php

```diff
@@ -197,21 +197,20 @@ public function loadData() {
 	private function getUserLinks( $user ) {
 
 		$userIsBlocked = $this->wg->User->isBlocked( true, false );
-		$oEncUserName = urlencode( $user->getName() );
+		$username = $user->getName();
 		$links = [
 			0 => "",
 			1 => Linker::link(
-				Title::newFromText( 'Contributions', NS_SPECIAL ),
-				$this->wg->Lang->ucfirst( wfMsg( 'contribslink' ) ),
-				[ 'target' => $oEncUserName]
+				SpecialPage::getSafeTitleFor( 'Contributions', $username ),
+				$this->wg->Lang->ucfirst( wfMsg( 'contribslink' ) )
 			),
 		];
 
 		if ( !empty( $this->wg->EnableWallExt ) ) {
```

Comment:
We could rely on `$user->getTalkPage()` to produce the correct talk page link, and generate the message key based on namespace to avoid a dependency on Wall here - this would also allow us to simplify the logic.